bin/build

```diff
@@ -5,16 +5,40 @@ set -o pipefail
 layers_dir=$1
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 echo -e 'launch = true' > $layers_dir/system-function.toml
+node_modules_dir=$layers_dir/node_modules
 system_dir=$layers_dir/system-function
+
+# create the caching layer for our node_modules
+touch "$node_modules_dir.toml"
+mkdir -p "$node_modules_dir"
+
+# where system function source code will be located and built
 mkdir -p $system_dir
 
 echo "---> Evergreen Node System Function Buildpack"
 
 # COPY and install node/npm on into a layer
 cp -R $DIR/../src/* $system_dir
-pushd $system_dir
-npm install
-popd
+
+local_lock_checksum=$(sha256sum "$system_dir/package-lock.json" | cut -d " " -f 1)
+cached_lock_checksum=$(yj -t < "$layers_dir/node_modules.toml" | jq -r ".metadata.package_lock_checksum")
```

Comment:
`jq` and `yj` don't exist in our stack image by default. Will they be present in the path like this? I know the `nodejs-npm-buildpack` installs them somewhere but I'm surprised the `PATH` change in that buildpack results in them being available for subsequent buildpacks.

Reply:
At the moment, I believe this works because `heroku/nodejs-engine` runs first, placing `jq` and `yj` in that `PATH` as build envs.  Arguably, buildpacks should be able to run independently, but I don't if it's worth it given that they have such strong dependencies on `herkoku/nodejs-engine` and `heroku/nodejs-npm`.

Reply:
Hmm yeah. I worry a bit about how brittle this might be if the `heroku/nodejs-npm` stops updating `PATH` but I guess we can swing back around to this later if these don't end up in our base stack image (I think we should consider it).

Reply:
I agree.  Seems reasonable to revisit if needed. 👍